src/libcore/intrinsics.rs

```diff
@@ -1917,6 +1917,15 @@ extern "rust-intrinsic" {
     #[rustc_const_unstable(feature = "const_discriminant", issue = "69821")]
     pub fn discriminant_value<T>(v: &T) -> <T as DiscriminantKind>::Discriminant;
 
+    /// Returns the number of variants of the type `T` cast to a `usize`;
```

Comment:
The docs should clarify whether uninhabited variants count. E.g., what is the return value for `Option<!>` or `Result<!, !>`?

Reply:
(Sorry, this comment should have gone to the public function, not the intrinsic.)

Reply:
Ah - this is a very good point